### PR TITLE
mon: collect mon metdata as part of the election

### PR DESCRIPTION
**Pull request by liewegas**
Fixes http://tracker.ceph.com/issues/20434

### CODE REVIEW

File: src/mon/Monitor.cc [hunk 1]
```diff
@@ -4712,16 +4713,7 @@ void Monitor::handle_mon_metadata(MonOpRequestRef op)
...
-  map<int, Metadata> last_metadata;
-  if (!err) {
-    bufferlist::iterator iter = bl.begin();
-    ::decode(last_metadata, iter);
-    pending_metadata.insert(last_metadata.begin(), last_metadata.end());
-  }
+  pending_metadata[from] = std::move(m);
 
   MonitorDBStore::TransactionRef t = paxos->get_pending_transaction();
   bl.clear();
```
Comment (LiumxNL) [inline src/mon/Monitor.cc]: bl not declared

**Comment by liewegas** [comment]
fixed build and repushed

**Comment by liewegas** [comment]
build is fixed, test commit removed... ready for review and test!

**Comment by liewegas** [comment]
retest this please

**Comment by liewegas** [comment]
http://pulpito.ceph.com/sage-2017-07-07_06:22:09-rados-wip-sage-testing2-distro-basic-smithi/

passes tests...

**Pull request merged**